rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricSummaryBridge.java

```diff
@@ -346,29 +354,44 @@ public CompletableFuture<List<Location>> getImplementations(Position cursor) {
         return implementations.thenCompose(d -> d.lookup(cursor).get());
     }
 
-    public void calculateSummary() {
-        calculateSummary(false);
+    public InterruptibleFuture<Lazy<List<Diagnostic>>> calculateSummary(CompletableFuture<Versioned<ITree>> tree) {
+        return calculateSummary(false, tree);
     }
 
-    private void calculateSummary(boolean internal) {
+    private InterruptibleFuture<Lazy<List<Diagnostic>>> calculateSummary(boolean internal, CompletableFuture<Versioned<ITree>> tree) {
         logger.trace("Requesting Summary calculation for: {}", file);
-        var summary = InterruptibleFuture.flatten(lookupState.apply(file)
-            .getCurrentTreeAsync()
-            .thenApplyAsync(t -> contrib.summarize(file, t), exec)
```

Comment:
this `.thenApplyAsync` is now missing (as it's not in `calculate` anymore)? is that intended?